work/src/org/kuali/kfs/module/ar/document/service/impl/ContractsGrantsInvoiceDocumentServiceImpl.java

```diff
@@ -1891,6 +1901,60 @@ public void recalculateSourceAccountingLineTotals(ContractsGrantsInvoiceDocument
         }
     }
 
+    /**
+     * Checks the KFS-AR / ContractsGrantsInvoiceDocumentBatchStep / User parameter, or, if no value is found for that parameter, the KFS system user to see if they are the
+     * initiator of the document.  If the KFS-AR / ContractsGrantsInvoiceDocumentBatchStep / User parameter is set to a value of a principal who creates CINVs outside of batch
+     * - well, okay, that's an interesting business process, that also means that the results of this method may be incorrect.  But you can always override, right?
+     * @see org.kuali.kfs.module.ar.document.service.ContractsGrantsInvoiceDocumentService#isDocumentBatchCreated(org.kuali.kfs.module.ar.document.ContractsGrantsInvoiceDocument)
+     */
+    @Override
+    public boolean isDocumentBatchCreated(ContractsGrantsInvoiceDocument document) {
+        if (document.getInvoiceGeneralDetail().getAward().getAutoApproveIndicator()) {
+            final Principal batchJobInitiatorPrincipal = getContractsGrantsInvoiceBatchCreationUserPrincipal();
+            return StringUtils.equalsIgnoreCase(document.getFinancialSystemDocumentHeader().getInitiatorPrincipalId(), batchJobInitiatorPrincipal.getPrincipalId());
+        }
+        return false;
+    }
+
+    /**
+     * @see org.kuali.kfs.module.ar.document.service.ContractsGrantsInvoiceDocumentService#doesInvoicePassValidation(org.kuali.kfs.module.ar.document.ContractsGrantsInvoiceDocument)
+     */
+    @Override
+    public boolean doesInvoicePassValidation(final ContractsGrantsInvoiceDocument document) {
```

Comment:
a) My concern with calling the validation more directly is simply that the institution may have changed something.  I can't get any closer to the actual validation before I start running into that issue.  b) Yeah, Exception is handed to me straight from java.util.Concurrent.Callable; no way to change.  Thoughts?